utils/db_train_test_split_by_user.py

```diff
@@ -17,15 +21,27 @@
 df = df.drop("session_id", axis="columns")
 user_ids = df["user_id"].unique()
 
-shuffle(user_ids)
-df_first_half_by_user_id = df.loc[df['user_id'].isin(user_ids[:int(len(user_ids)/2)])]
-df_second_half_by_user_id = df.loc[df['user_id'].isin(user_ids[int(len(user_ids)/2):])]
+if config["NORMAL_SPLIT"] == 'y':
+    shuffle(user_ids)
+    df_first_part_by_user_id = df.loc[df['user_id'].isin(user_ids[:int(len(user_ids) / 100 * (config["RATIO_OF_EXISTING_USERS"]))])]
+    df_second_part_by_user_id = df.loc[df['user_id'].isin(user_ids[int(len(user_ids) / 100 * (config["RATIO_OF_EXISTING_USERS"])):])]
 
-# main_user_ids = [item for item in user_ids if item.startswith('m_')]  #for split by prefix m_
-# other_user_ids = [item for item in user_ids if not item.startswith('m_')]
+    user_ids = df_first_part_by_user_id.pop("user_id")
 
-# df_first_half_by_user_id = df.loc[df['user_id'].isin(main_user_ids)]
-# df_second_half_by_user_id = df.loc[df['user_id'].isin(other_user_ids)]
+    df_first_part_by_user_id.pop("_id")
+    X_train, X_test, Y_train, Y_test = train_test_split(df_first_part_by_user_id, user_ids, stratify=user_ids)
+    train = X_train.join(Y_train)
+    test = X_test.join(Y_test)
 
-db_other["train"].insert_many(df_first_half_by_user_id.to_dict('records'))
-db_other["test"].insert_many(df_second_half_by_user_id.to_dict('records'))
+    db_other["train"].insert_many(train.to_dict('records'))
+    db_other["test"].insert_many(test.to_dict('records'))
+    db_other["test"].insert_many(df_second_part_by_user_id.to_dict('records'))
+
+else:
+    imitated_user_ids = [item for item in user_ids if re.match(r'^m.*_', item)]  # for split by prefix m_
```

Comment:
Tento regex hlada len `m_`? To sa mi akosi nezda, ved tam je .* takze zoberie hocico co zacina `m` a obsahuje `_`...

Naco je toto rozdelenie aka je toho logika prosim?

Kazdopadne ja nemam moc rad regexy -  nevieme radsej vyuzit `str.startswith();` pripadne  `if 'm' in string`?

Reply:
Nehľadá len m_ ale hľadá m(cokolvek)_ sú tak označené datá ľudi, ktorý boli napodobňovaní
a teda logika je, ze do trainu da ludi ktory boli napodobnovany a do testu tych ostatných.

if 'm' in string by ti moc nefungovalo kedze tu mas len z nas troch ludi, ktory maju "m" v mene. a startswith by sice fungoval na terajsie data, no v buducnosti ich budeme znacit pravdepodobne lepsie, aby sme vedeli kto koho napodobnoval, lebo teraz sa to da zistit iba z timestamp